.github/workflows/v2-deploy-pr.yml

```
@@ -0,0 +1,130 @@
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.updated_demos' > updated_demos.json
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.deleted_demos' > deleted_demos.json
+
+          echo "pr_id=$(cat pr_id.txt)" >> $GITHUB_OUTPUT
+          echo "pr_ref=$(cat pr_ref.txt)" >> $GITHUB_OUTPUT
+          echo "pr_ref_name=$(cat pr_ref_name.txt)" >> $GITHUB_OUTPUT
+          echo "updated_demos=$(cat updated_demos.json)" >> $GITHUB_OUTPUT
+          echo "deleted_demos=$(cat deleted_demos.json)" >> $GITHUB_OUTPUT
+
+      - name: Set job outputs
+        if: github.event.workflow_run.event == 'pull_request' && steps.build_context.outputs.result != ''
+        id: set_job_outputs
+        run: |
+          echo "pr_id=${{ steps.pr_info.outputs.pr_id }}" >> $GITHUB_OUTPUT
+          echo "pr_ref=${{ steps.pr_info.outputs.pr_ref }}" >> $GITHUB_OUTPUT
+          echo "pr_ref_name=${{ steps.pr_info.outputs.pr_ref_name }}" >> $GITHUB_OUTPUT
+          echo "updated_demos=${{ steps.pr_info.outputs.updated_demos }}" >> $GITHUB_OUTPUT
+          echo "deleted_demos=${{ steps.pr_info.outputs.deleted_demos }}" >> $GITHUB_OUTPUT
+    outputs:
+      pr_id: ${{ steps.set_job_outputs.outputs.pr_id }}
+      pr_ref: ${{ steps.set_job_outputs.outputs.pr_ref }}
+      pr_ref_name: ${{ steps.set_job_outputs.outputs.pr_ref_name }}
+      updated_demos: ${{ steps.set_job_outputs.outputs.updated_demos }}
+      deleted_demos: ${{ steps.set_job_outputs.outputs.deleted_demos }}
+
+  # Step 2: Deploy the demos to SWC
+  deploy-preview-demos:
+    if: github.event.workflow_run.event == 'pull_request' && needs.prepare-build-context.result == 'success'
+    uses: ./.github/workflows/v2-deploy-demos.yml
+    needs: prepare-build-context
+    with:
+      # TODO: Update SWC environment to "swc-prod"
+      environment: 'swc-staging' 
+      artifact-name: demo-build-${{ needs.prepare-build-context.outputs.pr_ref }}
+      preview: true
+    secrets: inherit
+  
+  # Step 3: Create a comment on the PR with the demo links
+  generate-comment:
+    if: github.event.workflow_run.event == 'pull_request' && needs.prepare-build-context.outputs.pr_id != ''
+    runs-on: ubuntu-latest
+    needs: prepare-build-context
+    steps:
+      - name: Create markdown comment from demo names
+        id: generate-markdown
+        run: |
+          demos="${{ needs.prepare-build-context.outputs.updated_demos }}"
+          
+          comment="### Preview(s) are ready! :tada:\n"
+          comment+="<details>\n"
+          comment+="<summary>Toggle to view preview links</summary>\n"
+          comment+="\n"
+          # TODO: Switch to prod once testing is complete
+          for demo in $demos; do
+            comment+="- [$demo](https://staging.pennylane.ai/qml/demos/$demo?preview=true)\n"
+          done
+      
+          comment+="\n"
+          comment+="</details>"
+
+          echo "::set-output name=markdown::$comment"
```

I believe `::set-output` is deprecate (or removed)
```suggestion
          echo "markdown=$comment" >> $GITHUB_OUTPUT
```

---

🤦🏻